moveit_simple/src/moveit_simple.cpp

```diff
@@ -443,66 +448,139 @@ bool Robot::jointInterpolation(const std::unique_ptr<TrajectoryPoint> & traj_poi
   }
   else
   {
-    current_point = traj_point->toJointTrajPoint(*this, IK_TIMEOUT, prev_point);
-    if (!current_point)
+    target_point = traj_point->toJointTrajPoint(*this, IK_TIMEOUT, prev_point);
+    if (!target_point)
     {
       ROS_WARN_STREAM("Failed to convernt joint point - this shouldn't happen");
       return false;
     }
   }
 
-  points.push_back(toJointTrajPtMsg(*current_point));
-  ROS_INFO_STREAM("Appending trajectory point, size: " << points.size());
+  unsigned int points_added = 0;
+  for (std::size_t i=1; i<=num_steps+1; ++i)
+  {
+    double t = (double)i / (double)(num_steps+1);
+    std::unique_ptr<JointTrajectoryPoint> new_point;
+    interpolate(prev_traj_point,target_point,t,new_point);
+    if(new_point)
+    {
+      points.push_back(toJointTrajPtMsg(*new_point));
+      ROS_INFO_STREAM("Appending trajectory point, size: " << points.size());
+      points_added++;
+      ROS_INFO_STREAM( points_added << " points among " << (num_steps+1) <<
+                      " successfully interpolated for " << traj_point->name());
+    }else{
+       ROS_WARN_STREAM("Conversion to joint trajectory failed at " << points_added << " among "
+                        << (num_steps+1) << "points for " << traj_point->name()
+                        << ". Clearing points added in unsuccessful attempt");
+      for (std::size_t j=0; j<points_added; ++j)
```

Comment:
This makes for an awkward method.  Can you think of a better way to write this method that doesn't require you to walk back failures?